src/olympia/devhub/tests/test_views_submit.py

```diff
@@ -119,11 +120,25 @@ def test_set_read_dev_agreement_error(self):
 
     def test_read_dev_agreement_skip(self):
         after_agreement_last_changed = (
-            UserProfile.last_developer_agreement_change + timedelta(days=1))
+            datetime(2018, 1, 1) + timedelta(days=1))
         self.user.update(read_dev_agreement=after_agreement_last_changed)
         response = self.client.get(reverse('devhub.submit.agreement'))
         self.assert3xx(response, reverse('devhub.submit.distribution'))
 
+    def test_read_dev_agreement_set_to_future(self):
+        set_config('last_dev_agreement_change_date', '2099-31-12 00:00')
```

Comment:
This is an invalid date (Y-D-M instead of Y-M-D), `strptime` will always raise `ValueError` on it.

Reply:
Note that it's good to have a test covering an invalid date, so you should add a couple tests for this case